src/hdusd/mx_nodes/node_tree.py

```diff
@@ -50,6 +52,17 @@ def output_node_volume(self):
         return next((node for node in self.nodes
                      if node.bl_idname == 'hdusd.MxNode_STD_volumematerial'), None)
 
+    def safe_call(self, op, *args, **kwargs):
+        """This function prevents call of self.update() during calling our function"""
+        if self._is_safe_call:
+            return op(*args, **kwargs)
```

Comment:
technically couldn't the call come between the if and return?  There must be some other way to do this with a multithread lock  it just seems the logic here is strange.

Reply:
Renamed to more related: safe_call -> no_update_call, _is_safe_call -> _do_update.